### DISCUSSION TITLE
Global is not defined

### DISCUSSION THREAD

**Discussion by bullypb23** [gatsbyjs/gatsby#30456] [Help]
## Description

Hi, I am trying to connect my Gatsby app with Looker instance, and when I try to do that I get this error. I believe it has something to do with @looker/chatty package I am using but can't figure out why.
![image](https://user-images.githubusercontent.com/55666396/112456258-e1374180-8d5a-11eb-8a94-37a7adebe6ab.png)

### Expected result

It should connect my app to my Looker instance, instead, it gives me this error.

### Environment

System:
    OS: macOS 11.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.19.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.90
    Firefox: 85.0.1
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.0.1 => 3.1.2
    gatsby-plugin-create-client-paths: ^3.0.0 => 3.1.0
    gatsby-plugin-gatsby-cloud: ^2.0.0 => 2.1.0
    gatsby-plugin-google-fonts: ^1.0.1 => 1.0.1
    gatsby-plugin-image: ^1.0.0 => 1.1.2
    gatsby-plugin-manifest: ^3.0.0 => 3.1.0
    gatsby-plugin-material-ui: ^2.1.10 => 2.1.10
    gatsby-plugin-offline: ^4.0.0 => 4.1.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.1.0
    gatsby-plugin-sharp: ^3.0.0 => 3.1.2
    gatsby-plugin-typescript: ^3.1.0 => 3.1.0
    gatsby-source-filesystem: ^3.0.0 => 3.1.0
    gatsby-transformer-sharp: ^3.0.0 => 3.1.0
  npmGlobalPackages:
    gatsby-cli: 3.0.0

**Comment by LekoArts** [comment]
Global is not available during SSR so you'll need to check against that: https://www.gatsbyjs.com/docs/debugging-html-builds/

Maybe webpack also removed a polyfill for this, too: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#webpack-5-node-configuration-changed-nodefs-nodepath-

**Comment by kelvindecosta** [comment]
This error happens because of `gatsby-plugin-image`. Refer to #30094 for more information.

As @LekoArts has stated, this is because of the webpack configuration.

I've tried adding the `node: { global: true }` polyfill in the webpack configuration, using `gatsby-node` API but this didn't seem to solve the issue.

**Reply by bullypb23** [reply]
I don't think that problem is with `gatsby-plugin-image`. I tried also adding `node: { global: true }` but I still get the same error. I believe that problem is with `@looker/chatty` package I am using because if I comment it out it works.

**Reply by kelvindecosta** [reply]
Have you tried commenting out `gatsby-plugin-image` in the list of plugins in `gatsby-config.js`?
This seems to remove the error for me.

**Reply by bullypb23** [reply]
I tried now and it works. I would never think it has something to do with `gatsby-plugin-image`. Thank you very much @kelvindecosta !

**Reply by ascorbic** [reply]
I have a PR open that fixes this. I strongly recommend that you don't remove the plugin from the config, as you'll lose a lot of the performance optimisations.
